### DISCUSSION TITLE
plot global variable with nans

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#5705] [Q&A] [answered]
The default `.plot()` function seems to struggle with global variables that have nans.

Here's is a file with u10 (u component of wind at 10 m) and swh (significant wave height)

[out.nc.zip](https://github.com/pydata/xarray/files/6984382/out.nc.zip)

```
import xarray as xr

ds = xr.open_dataset("out.nc")

ds["u10"].plot()
```

<img width="542" alt="Screen Shot 2021-08-13 at 2 06 14 PM" src="https://user-images.githubusercontent.com/17162724/129401089-83c8cdb3-d851-45e4-bb72-35818073d7d1.png">

```
ds["swh"].plot()
```
<img width="460" alt="Screen Shot 2021-08-13 at 2 06 58 PM" src="https://user-images.githubusercontent.com/17162724/129401160-4beffe30-2905-4c55-8b2d-7da8de8d8bd2.png">

AFAICT only difference is swh has NaNs

Same behavior with pcolomesh

```
ds["swh"].plot.pcolormesh("longitude", "latitude")
```

<img width="534" alt="Screen Shot 2021-08-13 at 2 10 20 PM" src="https://user-images.githubusercontent.com/17162724/129401533-2adcdbb3-b652-4ba6-9afc-b14f68dae69e.png">

**Comment by keewis** [comment, marked as answer]
there seems to be something wrong with your data:
```python
ds.swh.where(ds.swh.notnull(), drop=True).plot()
```
results in the expected plot, but the latitude and longitude coordinates cover only 0 to 180 degrees on the northern hemisphere (despite the plot covering the whole globe).

Maybe there's something wrong with the processing steps producing this dataset?

**Reply by raybellwaves** [reply]
thanks. I'll take a look.

FYI source is similar to...

```
import cdsapi

c = cdsapi.Client()

c.retrieve(
    "reanalysis-era5-single-levels",
    {
        "product_type": "reanalysis",
        "variable": [
            "10m_u_component_of_wind",
            "significant_height_of_combined_wind_waves_and_swell",
        ],
        "year": "2020",
        "month": "01",
        "day": "01",
        "time": "00:00",
        "format": "netcdf",
    },
    "/tmp/download.nc",
)
```

**Reply by raybellwaves** [reply]
Think I got it. Wind and waves are at different resolution in ERA5 so think it does something funny when you request them together.
